web/src/engine/websites/CoroCoro.ts

```diff
@@ -25,4 +63,69 @@ export default class extends DecoratableMangaScraper {
     public override get Icon() {
         return icon;
     }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL('/rensai', this.URI)), 'script');
+        const json = this.FindAndParseJSON(scripts, 'totalChapterLikes');
+        const { weekdays: { mon, tue, wed, thu, fri, sat, sun } } = this.LocateObjectWithNamedProperty<JSONMangas>(json, 'weekdays');
+        return [mon, tue, wed, thu, fri, sat, sun].reduce((accumulator: Manga[], day) => {
+            const mangas = day.map(manga => new Manga(this, provider, `/title/${manga.id}`, manga.name));
+            accumulator.push(...mangas);
+            return accumulator;
+        }, []);
+
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL(`${manga.Identifier}`, this.URI)), 'script');
+        const json = this.FindAndParseJSON(scripts, 'omittedMiddleChapters');
+        const { chapters: { earlyChapters, omittedMiddleChapters, latestChapters } } = this.LocateObjectWithNamedProperty<JSONChapters>(json, 'chapters');
+        return [earlyChapters, omittedMiddleChapters, latestChapters].reduce((accumulator: Chapter[], category) => {
+            const chapters = category.map(chapter => new Chapter(this, manga, `/chapter/${chapter.id}`, chapter.title));
+            accumulator.push(...chapters);
+            return accumulator;
+        }, []);
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page<CryptoParams>[]> {
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL(`${chapter.Identifier}`, this.URI)), 'script');
+        const json = this.FindAndParseJSON(scripts, 'viewerSection');
+        if (!json) {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+        const { viewerSection: { pages } } = this.LocateObjectWithNamedProperty<JSONPages>(json, 'viewerSection');
+        return pages.map(page => new Page<CryptoParams>(this, chapter, new URL(page.src), page.crypto));
+    }
+
+    public override async FetchImage(page: Page<CryptoParams>, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal, true);
+        const cryptoParams = page.Parameters;
+        if (!cryptoParams?.key) return blob;
```

Comment:
Per type definition this field cannot be undefined.
Either update the type definition and mark it as nullable, or remove this condition.